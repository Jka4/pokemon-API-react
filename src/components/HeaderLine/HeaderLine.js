import React from 'react';
import SearchInput from "../SearchInput/SearchInput";
import Player from "../Player/Player";
import './styles/style.scss';

const HeaderLine = () => {
  return (
    <div className='topLine'>
      <Player />
      <SearchInput />
    </div>
  );
}

export default HeaderLine;
