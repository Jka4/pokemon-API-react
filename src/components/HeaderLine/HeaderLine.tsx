import React from "react";
import { NavLink } from "react-router-dom";

import store from "../../Store/store";

import Search from "../Search/Search";
import Player from "../Player/Player";

import AppBar from "@material-ui/core/AppBar";
import { Button } from "@material-ui/core";
import SvgIcon from "@material-ui/core/SvgIcon";

import "./styles/style.scss";

interface HeaderLineProps {
  pathname?: string;
  isMainPage?: any;
}

const HeaderLine: React.FC<HeaderLineProps> = ({
  pathname,
}: HeaderLineProps) => {
  const isMainPage = pathname === "/";

  function HomeIcon({ props }: any) {
    return (
      <SvgIcon {...props}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    );
  }

  return (
    <>
      <AppBar position="static" className="topLine">
        <div className="buttonsRow">
          <Player />

          {!isMainPage && (
            <NavLink
              to="/"
              onClick={() => store.dispatch({ type: "CLEAR_DETAILS_PAGE" })}
            >
              <Button variant="contained" className="toHome" color="secondary">
                <HomeIcon />
              </Button>
            </NavLink>
          )}
        </div>
        <Search />
      </AppBar>
    </>
  );
};

export default HeaderLine;
