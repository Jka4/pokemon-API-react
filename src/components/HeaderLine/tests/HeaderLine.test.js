import React from 'react';
import HeaderLine from '@HeaderLine';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from "react-router-dom";

it('HeaderLine renders correctly', async () => {
	const tree = renderer
		.create(<Router><HeaderLine /></Router>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
