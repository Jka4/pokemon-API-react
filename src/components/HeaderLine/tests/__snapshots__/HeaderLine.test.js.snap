// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLine renders correctly 1`] = `
<div
  className="topLine"
>
  <a
    aria-current="page"
    className="backToMainPage active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
     Back 
  </a>
  <div
    className="player"
  >
    <button
      className="playButton"
      onClick={[Function]}
    >
      PLAY
    </button>
  </div>
  <div
    className="search"
  >
    <input
      aria-label="Search"
      autoComplete="off"
      id="searchInput"
      name="searchInput"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Pickachu, 123, 900g"
      type="search"
    />
    <div
      className="searchResults"
    >
      <ul
        className="searchList"
      />
    </div>
  </div>
</div>
`;
